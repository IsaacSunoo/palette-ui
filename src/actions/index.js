export const setError = error => ({
  type: "SET_ERROR",
  error
});

export const setLoading = loading => ({
  type: "SET_LOADING",
  loading
});

export const setPalette = palette => ({
  type: "SET_PALETTE",
  palette
});

export const setLock = colorID => ({
  type: "SET_LOCK",
  colorID
});

export const setModal = status => ({
  type: "SET_MODAL",
  status
});

export const setProjects = projects => ({
  type: "SET_PROJECTS",
  projects
});
