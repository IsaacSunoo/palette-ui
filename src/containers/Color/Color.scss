.Color {
  display: flex;
  width: 20%;
  justify-content: flex-end;
  align-items: center;
  flex-direction: column;

  .LockBtn {
    margin: 0;

    button {
      background: none;
      border: none;
      outline: none;
      cursor: pointer;

      img {
        width: 100%;
        max-width: 50px;
      }
    }
  }

  .ColorCode {
    margin: 0;
  }

}
