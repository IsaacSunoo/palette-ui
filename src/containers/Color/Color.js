import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import * as actions from '../../actions';

import lockIcon from '../../media/icons/lock-alt-solid.svg';
import unlockIcon from '../../media/icons/unlock-alt-solid.svg';

export class Color extends Component {

  handleLock = id => {
    const { setLock } = this.props;
    setLock(id);
  }

  render(){
    let { color, id, locked } = this.props;
    let lockStatus = locked ? lockIcon : unlockIcon;

    return (
      <div className='Color' style={{ backgroundColor: this.props.color }}>
        <div className='LockBtn'>
          <button className='handleLockClick' onClick={() => this.handleLock(id)}>
            <img src={lockStatus} className='Lock' alt={color} />
          </button>
        </div>
        <div className='ColorCode'>
          <h3>{color}</h3>
        </div>
      </div>
    )
  }
}

export const mapDispatchToProps = dispatch => ({
  setLock: data => dispatch(actions.setLock(data)),
});

Color.propTypes = {
  setLock: PropTypes.func
}

export default connect(null, mapDispatchToProps)(Color);
