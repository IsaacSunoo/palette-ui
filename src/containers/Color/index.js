import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import * as actions from '../../actions/index';

import lockIcon from '../../media/icons/lock-alt-solid.svg';
import unlockIcon from '../../media/icons/unlock-alt-solid.svg';

export class Color extends Component {

  handleLock = () => {
    this.props.setLock(['1'])
  }

  render(){
    let { color } = this.props
    let lockStatus = 2 < 1 ? lockIcon : unlockIcon

    return (
      <div className="Color" style={{backgroundColor: this.props.color}}>
        <div className="LockBtn">
          <button onClick={() => this.handleLock()}>
            <img src={lockStatus} className="Lock" alt={color} />
          </button>
        </div>
        <div className="ColorCode">
          <h3>{color}</h3>
        </div>
      </div>
    )
  }
}


export const mapStateToProps = (state) => ({

})

export const mapDispatchToProps = (dispatch) => ({
  setLock: (data) => dispatch(actions.setLock(data)),
})

Color.propTypes = {
  setLock: PropTypes.func
}

export default connect(mapStateToProps, mapDispatchToProps)(Color);
