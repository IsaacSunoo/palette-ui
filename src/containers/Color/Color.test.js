import React from 'react';
import { shallow, mount } from 'enzyme'
import { Color, mapDispatchToProps } from './Color';

import * as actions from '../../actions/index';

describe('Color', () => {

  describe('Color Component', () => {
    let wrapper;
    let mockSetLock = jest.fn()

    beforeEach(() => {
      wrapper = shallow(<Color setLock={mockSetLock}/>)
    })

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    })

  });

  describe('mapDispatchToProps', () => {

    it('should call dispatch for fetchAllProjects', () => {
      //Setup
      const mockData = true;
      const mockDispatch = jest.fn();
      const actionToDispatch = actions.setLock(mockData)
      //Execution
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.setLock(mockData);
      //Expectation
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })

  });

});
