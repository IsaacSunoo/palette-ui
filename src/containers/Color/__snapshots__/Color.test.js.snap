// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color Color Component should match the snapshot 1`] = `
<div
  className="Color"
  style={
    Object {
      "backgroundColor": undefined,
    }
  }
>
  <div
    className="LockBtn"
  >
    <button
      className="handleLockClick"
      onClick={[Function]}
    >
      <img
        className="Lock"
        src="unlock-alt-solid.svg"
      />
    </button>
  </div>
  <div
    className="ColorCode"
  >
    <h3 />
  </div>
</div>
`;
